core/include/traccc/definitions/qualifiers.hpp

```diff
@@ -31,3 +31,16 @@
 #else
 #define TRACCC_ALIGN(x) alignas(x)
 #endif
+
+#if __has_builtin(__builtin_assume)
+#define TRACCC_ASSUME(cond) __builtin_assume(cond)
+#else
+#define TRACCC_ASSUME(cond)
+#endif
```

Comment:
```suggestion
#if __GNUC__ < 10
#define TRACCC_ASSUME(cond)
#elif __has_builtin(__builtin_assume)
#define TRACCC_ASSUME(cond) __builtin_assume(cond)
#else
#define TRACCC_ASSUME(cond)
#endif
```

GCC < 10 is unaware of `__has_builtin`

Reply:
The recommended syntax is `#if defined __has_builtin` as per https://gcc.gnu.org/onlinedocs/cpp/_005f_005fhas_005fbuiltin.html

Reply:
I'm sorry to be this anal, but this is not really a qualifier. Perhaps we could create a new file `builtins.hpp` or `hints.hpp` (open for discussion) for this?

Reply:
If it's `hints.hpp` then we can put the new pragma unroll in there as well.

Reply:
Agreed!